chapters/teorija-stevil/teorija-stevil.tex

```diff
@@ -0,0 +1,168 @@
+Izračunamo c = a % b (ostanek pri deljenju)
+Ponavljamo, dokler je c različen od 0
+    za novi a nastavimo star b
+    za novi b nastavimo star c
+    za novi c nastavimo (novi a) % (novi b)
+Končni odgovor je b.
+\end{verbatim}
+Tak opis lahko tudi enostavno zapišemo kot funkcijo v C++:
+%#insert python3 style.py < chapters/teorija-stevil/gcd.cpp
+
+\section{Eratostenovo rešeto}
+
+\emph{Praštevilo} je tako število, ki je deljivo le z $1$ in samo s seboj.
+Praštevil je neskončno, najmanjša od njih so $2, 3, 5, 7, 11, 13, \ldots$.
+Praštevila so v matematiki zelo pomembna, med drugim zaradi naslednje trditve,
+ki se ji reče \emph{osnovni izrek teorije števil}.
+
+\vspace{0.5cm}
+
+\textbf{Izrek:} Vsako število lahko na enoličen način zapišemo kot produkt
+praštevil.
+
+\begin{examples}
+  Izrek nam pove, da lahko vsakemu številu poiščemo praštevilski razcep.
+  Poglejmo si praštevilski razcep števila $6552$.
+  To število je deljivo z $2$, pri deljenju dobimo količnik $3276$.
+  Ta količnik lahko spet delimo z $2$ in dobimo $1638$.
+  Ko še tretjič delimo z $2$, dobimo rezultat $819$, to število pa ni več
+  deljivo z $2$.
+  Lahko ga še dvakrat delimo s $3$; sprva dobimo $273$, nato $91$.
+  Naposled lahko $91$ delimo s $7$ in dobimo $13$.
+  Praštevilski razcep števila $6552$ je tako
+  \[
+	6552 = 2^3 \cdot 3^2 \cdot 7 \cdot 13.
+  \]
+\end{examples}
+
+V primeru smo videli, da se v praštevilskem razcepu lahko kakšno praštevilo
+ponavlja, kakšno pa se sploh ne pojavi.
+Če želimo razcep poiskati s programom, ali pa narediti kaj drugega s praštevili,
+pa jih moramo prvo poiskati.
+Spoznali bomo en algoritem za iskanje praštevil, to je Eratostenovo rešeto.
+
+Recimo, da želimo poiskati vsa praštevila do neke zgornje meje $N$.
+Sprva si napišemo vsa števila med $2$ in $N$ ($1$ namreč ni praštevilo);
+
+\includegraphics[width=\linewidth]{chapters/teorija-stevil/slike/reseto1}
+
+Vemo, da je število $2$ praštevilo.
+Večkratniki števila $2$ zato niso praštevila, ker so deljivi z $2$; torej jih
+prečrtamo.
+
+\includegraphics[width=\linewidth]{chapters/teorija-stevil/slike/reseto2}
+
+Število $3$ je praštevilo, torej tudi večkratniki $3$ niso praštevila.
+
+\includegraphics[width=\linewidth]{chapters/teorija-stevil/slike/reseto3}
+
+Prišli so do števila $4$, ki je prečrtano, zato vemo, da ni praštevilo.
+Ker je deljivo z $2$, smo vse večkratnike $4$ (ki so tudi večkratniki $2$), že
+prečrtali, torej nam ni treba posebej črtati večkratnikov $4$.
+
+Število $5$ je praštevilo (ker ni prečrtano), torej moremo prečrtati njegove
+večkratnike.
+
+\includegraphics[width=\linewidth]{chapters/teorija-stevil/slike/reseto4}
+
+Število $6$ je prečrtano, tako da ga preskočimo.
+
+Postopek bi lahko nadaljevali, dokler ne pridemo do $26$ (oz.~do želene zgornje
+meje), vendar nam ni treba.
+Velja namreč naslednje; če je število $M$ produkt dveh števil, $M = A B$, potem
+je vsaj eno od teh števil manjše od ali enako veliko kot $\sqrt{M}$.
```

Comment:
Kaj je koren, je to smiselno razlagat?

Reply:
Kakor sem jih na uri vprašal, so vedeli.